app/org/sagebionetworks/bridge/config/BridgeSpringConfig.java

```diff
@@ -199,6 +199,9 @@ public JedisPool jedisPool(BridgeConfig config) {
         // configure Jedis pool
         JedisPoolConfig poolConfig = new JedisPoolConfig();
         poolConfig.setMaxTotal(config.getPropertyAsInt("redis.max.total"));
+        poolConfig.setTestOnBorrow(true);
```

Comment:
As far as I can tell, it checks that the pool thread has an active connection with the Redis server when it first borrows a connection from the pool and again when it returns the connection to the pool. It's not clear to me what happens if the test fails (although in the worst case, we'd be fast failing here instead of failing later in the Redis call).

Presumably, there is a small overhead, since we'll need to ping the Redis server to get the connection.

After researching possible Redis connection errors, I found this thread https://github.com/xetorthio/jedis/issues/91 which recommends setting both testOnBorrow and testOnReturn. The thread is about 4 years old, so it might be outdated, but there isn't any newer documentation.

I think it's worth trying, since there are still a lot of JedisConnectionExceptions in our logs.